### PR TITLE
Add board "Lilygo TTGO T7 V1.5"

### PR DESCRIPTION
**Pull request by kaessert**
This PR adds support for the Lilygo TTGO T7 V1.5 baord. It's my first pr of that kind so please be gentle with me :)

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by kaessert** [comment]
Thank you so much for your review @nordicjm ! I addressed the comments.

File: boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2024 Tobias Kässer <t.kaesser@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_TTGO_T7V1_5
+	select SOC_ESP32
+
+config ESP_WIFI_MAX_THREAD_PRIORITY
+    int "ESP WiFi Maximum Thread Priority"
+    default 3
```
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig:9]: Don't know what this is but it can go

File: boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2024 Tobias Kässer <t.kaesser@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_TTGO_T7V1_5
+	select SOC_ESP32
```
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig:5]: this does not go in this file, it should be `Kconfig.ttgo_t7v1_5`

File: boards/lilygo/ttgo_t7v1_5/doc/img/ttgo_t7v1_5.webp
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/doc/img/ttgo_t7v1_5.webp:1]: Image is way too big

File: boards/lilygo/ttgo_t7v1_5/Kconfig [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2024 Tobias Kässer <t.kaesser@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 65535 if WIFI && BT
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/Kconfig:6]: @kaessert, let this be only `default 4096` instead. You can check other esp32-based boards as reference.

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,231 @@
+.. _ttgo_t7v1_5:
+
+Lilygo TTGO T7 V1.5
+#################
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:4]: Change the 4 lines above to:
```
.. zephyr:board:: ttgo_t7v1_5
```

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+based on the Espressif ESP32-WROVER-E module.
+
+It features the following integrated components:
+- ESP32 chip (240MHz dual core, 520KB SRAM, Wi-Fi, Bluetooth)
+- on board antenna
+- Micro-USB connector for power and communication
+- JST GH 2-pin battery connector
+- LED
+
+.. figure:: img/ttgo_t7v1_5.webp
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:19]: You can remove thiswhoe figure entry entry here (documentation will automatically add it). Please, decrease image size as well.

File: boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_procpu.dts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+&timer3 {
+	status = "okay";
+};
+
+&trng0 {
+	status = "okay";
+};
+
+&flash0 {
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_procpu.dts:95]: As you included `#include <espressif/esp32/esp32_wrover_e_n4r8.dtsi>`, you can remove all this `flash0` entry from here.

File: boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_appcpu.dts [hunk 7]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32/esp32_appcpu.dtsi>
+
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_appcpu.dts:9]: Pease, add 
`#include <espressif/partitions_0x1000_amp.dtsi>` 
and remove `flash0` entry below.

File: boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig (hunk 1 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig:9]: Without it i'm getting these kind of errors on build:
```
/home/cite/emb-test/projects/lora-receiver-zephyr/prj.conf:5: warning: attempt to assign the value '6' to the undefined symbol ESP_WIFI_MAX_THREAD_PRIORITY
```

File: boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig (hunk 2 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/Kconfig.defconfig]: done

File: boards/lilygo/ttgo_t7v1_5/Kconfig (hunk 3 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/Kconfig]: done

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst (hunk 4 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/doc/index.rst]: done

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst (hunk 5 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/doc/index.rst]: done

File: boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_procpu.dts (hunk 6 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_procpu.dts]: done

File: boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_appcpu.dts (hunk 7 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/ttgo_t7v1_5_esp32_appcpu.dts:10]: done

**Comment by kaessert** [comment]
Thank you so much for your patience :pray:

File: boards/lilygo/ttgo_t7v1_5/Kconfig.ttgo_t7v1_5 [hunk 8]
```diff
@@ -0,0 +1,5 @@
+config BOARD_TTGO_T7V1_5
+	select SOC_ESP32
+	select SOC_ESP32_PROCPU if BOARD_TTGO_T7V1_5_ESP32_PROCPU
+	select SOC_ESP32_APPCPU if BOARD_TTGO_T7V1_5_ESP32_APPCPU
+
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/Kconfig.ttgo_t7v1_5:5]: Would you remove this extra line here?

Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/Kconfig.ttgo_t7v1_5]: sure thing, done

**Comment by kaessert** [comment]
@nordicjm Sure, squashed

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+Functional Description
+**********************
+This board is based on the ESP32-WROVER-E module with 4MB of flash (there
+are models 16MB as well), WiFi and BLE support. It has a Micro-USB port for
+programming and debugging, integrated battery charging and an on-board antenna.
+
+Connections and IOs
+===================
+
+The ``ttgo_t7v1_5`` board target supports the following hardware features:
```
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:25]: needs proper board targets (appcpu and procpu)

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+   :board: ttgo_t7v1_5
+   :goals: build
+
+The usual ``flash`` target will work with the ``ttgo_t7v1_5`` board
+configuration. Here is an example for the :zephyr:code-sample:`hello_world`
+application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: ttgo_t7v1_5
```
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:174]: these will not work

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+| BLE       | on-chip    | bluetooth_esp32  |
++-----------+------------+------------------+
+| Flash     | on-chip    | flash_esp32      |
++-----------+------------+------------------+
+
+Start Application Development
+*****************************
+
+Before powering up your Lilygo ``ttgo_t7v1_5``, please make sure that the board is in good
+condition with no obvious signs of damage.
```
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:53]: is this really needed?

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+   :app: samples/basic/blinky
+   :board: ttgo_t7v1_5
+   :goals: build
+
+To build the bluetooth beacon sample:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/bluetooth/beacon
+   :board: ttgo_t7v1_5
```
Comment (nordicjm) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:213]: will not work

Comment (sylvioalves) [reply boards/lilygo/ttgo_t7v1_5/doc/index.rst:213]: I was just writing this..

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+
+The sysbuild makes possible to build and flash all necessary images needed to
+bootstrap the board with the ESP32 SoC.
+
+To build the sample application using sysbuild use the command:
+
+.. zephyr-app-commands::
+   :tool: west
+   :app: samples/hello_world
+   :board: ttgo_t7v1_5
```
Comment (sylvioalves) [inline boards/lilygo/ttgo_t7v1_5/doc/index.rst:116]: `:board: ttgo_t7v1_5/esp32/procpu`

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst (hunk 11 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/doc/index.rst]: no, took it from docs from another lilygo board. removed.

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst (hunk 10 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/doc/index.rst]: should be fixed

**Comment by kaessert** [comment]
Found and fixed an error in my appcpu-defconfig. Tests should pass now.

**Review by kartben — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

File: boards/lilygo/ttgo_t7v1_5/doc/index.rst (hunk 9 above)
Comment (kaessert) [reply boards/lilygo/ttgo_t7v1_5/doc/index.rst]: figured it out, done

**Review by nordicjm — APPROVED** [review]

**Comment by kaessert** [comment]
Thanks for your support everybody!

**Pull request merged**